core/src/main/java/org/bitcoinj/net/discovery/DnsDiscovery.java

```diff
@@ -56,7 +56,7 @@ public DnsDiscovery(String[] dnsSeeds, NetworkParameters params) {
     }
 
     private static List<PeerDiscovery> buildDiscoveries(NetworkParameters params, String[] seeds) {
-        List<PeerDiscovery> discoveries = new ArrayList<PeerDiscovery>(seeds.length);
+        List<PeerDiscovery> discoveries = new ArrayList<PeerDiscovery>(seeds != null ? seeds.length : 0);
```

Comment:
Ah you're right; I was thinking that enhanced for loops would skip null variables. I'm assuming that the sending of the size parameter is for performance & security reasons, so I didn't want to remove it.